lib/finch/pool.ex

```diff
@@ -123,4 +120,19 @@ defmodule Finch.Pool do
       :closed
     end
   end
+
+  defp emit_checkout_event(conn) do
+    idle_time = System.monotonic_time() - conn.last_checkin
+    meta = telemetry_meta(conn, self())
+    Telemetry.event(:handle_checkout, %{idle_time: idle_time}, meta)
```

Comment:
Looks great! One alternative, so we emit fewer telemetry events, is to return the `idle_time` to the client, and include it as a measurement alongside `Telemetry.stop(:queue)`.

Reply:
I like that idea! I was not totally sure that I should add a new event just for this.